canonical/mimir-coordinator-k8s-operator
# First TLS instrumentation

This PR fixes: #15 

- Adds TLS config for the workers
- Adds TLS config for the NGINX workload in the coordinator.

And goes in tandem with:

- https://github.com/canonical/observability-libs/pull/72
- https://github.com/canonical/mimir-worker-k8s-operator/pull/21



## How to test it:

1. Deploy this bundle (You need to pack the coordinator and the worker)
```yaml
bundle: kubernetes
applications:
  ca:
    charm: self-signed-certificates
    channel: edge
    revision: 52
    scale: 1
    constraints: arch=amd64
  coord:
    charm: local:mimir-coordinator-k8s-21
    scale: 1
    constraints: arch=amd64
    storage:
      data: kubernetes,1,1024M
    trust: true
  worker:
    charm: local:mimir-worker-k8s-8
    scale: 1
    options:
      all: true
    constraints: arch=amd64
    storage:
      data: kubernetes,1,1024M
    trust: true
relations:
- - coord:mimir-cluster
  - worker:mimir-cluster
- - coord:certificates
  - ca:certificates
```
2. Check that the worker config file has the TLS settings in place: `juju ssh --container mimir worker/0 cat /etc/mimir/mimir-config.yaml`
3. Check the nginx config in the coordinator has SSL: `juju ssh --container nginx coord/0 cat /etc/nginx/nginx.conf`
